src/components/Sections/Sketches/ZipFileModal.js

```diff
@@ -0,0 +1,97 @@
+import React, { useRef, useState } from 'react';
+import { withNamespaces } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { Modal, Button } from 'semantic-ui-react';
+import { selectors as assets } from '../../../redux/modules/assets';
+import { isLanguageRtl } from '../../../helpers/i18n-utils';
+import { selectors as settings } from '../../../redux/modules/settings';
+import ImageGallery from 'react-image-gallery';
+import { imageGalleryItem } from '../../Pages/Unit/widgets/UnitMaterials/helper';
+import FallbackImage from '../../shared/FallbackImage';
+import { assetUrl } from '../../../helpers/Api';
+
+const ZipFileModal = ({ id, path }) => {
+  const [open, setOpen] = useState(false);
+
+  const getZipById = useSelector(state => assets.nestedGetZipById(state.assets));
+  const language   = useSelector(state => settings.getLanguage(state.settings));
+  const ref        = useRef();
+
+  const { data: { uniq, full } } = getZipById(id);
+
+  const uniqIdx  = uniq.findIndex(x => x.path === path);
+  const prev     = uniq?.[uniqIdx - 1] || null;
+  const startIdx = prev ? full.findIndex(x => x.path === prev.path) + 1 : 0;
+  const endIdx   = full.findIndex(x => x.path === path);
+
+  const items = full.slice(startIdx, endIdx + 1).map(imageGalleryItem);
+
+  const renderLeftNav = (onClick, disabled) => (
+    <Button
+      color="black"
+      size="tiny"
+      className="image-gallery-left-nav"
+      icon="chevron left"
+      disabled={disabled}
+      onClick={onClick}
+    />
+  );
+
+  const renderRightNav = (onClick, disabled) => (
+    <Button
+      color="black"
+      size="tiny"
+      className="image-gallery-right-nav"
+      icon="chevron right"
+      disabled={disabled}
+      onClick={onClick}
+    />
+  );
+
+  const renderFullscreenButton = (onClick, isFullscreen) => (
+    <Button
+      color="black"
+      size="tiny"
+      className="image-gallery-fullscreen-button"
+      icon={isFullscreen ? 'compress' : 'expand'}
+      onClick={onClick}
+    />
+  );
+
+  return (
+    <>
+      <FallbackImage
+        src={assetUrl(path.substring(8))}
```

Comment:
Feels fragile, do we do that to extract the UID?
Do we do that in all places?

Reply:
I was take it from prev code. From unzip service return only path and size of file. Its not take UID, its remove "/asserts" from image path